evo1/atb/entity.py

```diff
@@ -0,0 +1,173 @@
+        case ATBEntityID.CLINK:
+            max_hp = 100
+            if level >= 3:
+                attack = 26  # Claud's sword, lvl3
+            elif level >= 2:
+                attack = 21  # long sword, lvl2
+            elif level >= 1:
+                attack = 20  # long sword, lvl1
+            else:
+                attack = 8  # baby sword
+            if level >= 3:
+                defense = 6  # armor, lvl3
+            elif level >= 1:
+                defense = 5  # armor, lvl1
+            else:
+                defense = 0  # no armor
+            magic = 12
+            evade = 0
+        case ATBEntityID.KAERIS:
+            max_hp = 70
+            attack = 10
+            if level >= 2:
+                defense = 6  # lvl2
+            else:
+                defense = 5  # lvl1
+            if level >= 3:
+                magic = 6  # lvl3
+            else:
+                magic = 5  # lvl1
+            evade = 0
+        # Bosses
+        case ATBEntityID.KEFKAS_GHOST:
+            max_hp = 250
+            attack = 22
+            defense = 5
+            magic = 0
+            evade = 0
+        case ATBEntityID.ZEPHYROS:
+            max_hp = 250
+            # TODO: Stats
+            attack = 5
+            defense = 0
+            magic = 0
+            evade = 0
+        # Enemies
+        case ATBEntityID.SLIME:
+            max_hp = 12
+            attack = 5
+            defense = 0
+            magic = 0
+            evade = 0
+        case ATBEntityID.EMUK:
+            max_hp = 20
+            attack = 7
+            defense = 2
+            magic = 0
+            evade = 0
+        case ATBEntityID.SCAVEN:
+            max_hp = 11
+            attack = 7
+            defense = 2
+            magic = 0
+            evade = 0
+        case ATBEntityID.KOBRA:
+            max_hp = 15
+            attack = 10
+            defense = 10
+            magic = 0
+            evade = 30
+        case ATBEntityID.TORK:
+            max_hp = 15
+            attack = 18
+            defense = 17
+            magic = 0
+            evade = 0
+        case ATBEntityID.ZOOMBA:
+            max_hp = 30
+            attack = 10
+            defense = 5
+            magic = 0
+            evade = 0
+        case ATBEntityID.ATUIN:
+            max_hp = 50
+            attack = 15
+            defense = 15
+            magic = 0
+            evade = 0
+        case ATBEntityID.APIDYA:
+            max_hp = 19
+            attack = 16
+            defense = 5
+            magic = 0
+            evade = 25
+        # TODO: More stat blocks
+        case _:
+            max_hp = 0
+            attack = 0
+            defense = 0
+            magic = 0
+            evade = 0
+
+    return ATBEntityStats(
+        max_hp=max_hp,
+        cur_hp=max_hp,
+        attack=attack,
+        defense=defense,
+        magic=magic,
+        evade=evade,
+    )
+
+
+def atb_stats_from_memory(entity: BattleEntity) -> ATBEntityStats:
```

Comment:
In combat, we can read all stats from memory and pass these to the attack prediction.

Reply:
Lol, the thing from my last comment would have looked a lot like that. Lol